### ISSUE TITLE
harmo_process() error(s) using Rmonize_examples() objects

### ISSUE THREAD

**Issue by twey2** [maelstrom-research/Rmonize#97]
Running harmo_process() with Rmonize_examples() objects produces an ERROR in an algorithm that is written correctly in the DPE and produces some other issues. (Package versions: Rmonize_1.1.0.9106, madshapR_1.1.0.9111, fabR_2.1.0.9103)

Running the following script:
```
harmonized_dossier <- harmo_process(
  dossier,
  dataschema,
  data_proc_elem,
  harmonized_col_dataset = 'adm_study_id')
```
Produces the following processing error in dataset_study5:
![{080FDC5A-2C93-4A56-B064-AA209D50FC06}](https://github.com/user-attachments/assets/5dad9e14-269d-48f7-8ef1-31cf07455942)
![{88AB16AD-1C58-4477-9C42-434EA16518A0}](https://github.com/user-attachments/assets/bd7a5ab3-b382-4c42-b6ef-afc7fabf95fd)

The Mlstr_harmo::algorithm appears to be written correctly. The input variable appears to be formatted as a mixed variable (categorical and non-categorical values present).
![{D5A8478A-128C-43DB-ACBF-59C1E5ADA41D}](https://github.com/user-attachments/assets/b70b1920-903b-4638-a11f-cab72a1e421f)

The error is not reproduced if the algorithm script is run on the input variable directly in R.
![{59AEE055-B6C5-4FD6-BC39-A697F312DE26}](https://github.com/user-attachments/assets/902e9b7d-7ad6-4ecf-8389-f71556962c2d)

In addition, the harmonization process incorrectly prints a message about undetermined statuses (there are none).
![{91E6DB3C-83C1-4AC3-92AC-22CB11594129}](https://github.com/user-attachments/assets/41fdf3f7-1a01-46fc-bcfc-859d932dfb6d)

There are no warning messages about the processing error in dataset_study5. In the harmonized dossier, dataset_study5 is (correctly) empty.

**Comment by GuiFabre** [comment]
Hello @twey2
the bug has been corrected. The problem came from madshapR data_dict_zap_dataset() function which aims at separate the data dictionary from its associated dataset. An interesting case come with a factor variable (or haven_labelled variable in haven). If the the variable remains factor, then (technically) the data dictionary for its particular variable remains (the categories are levels/labels). If not, the column is a regular column, coerced with its associated valueType (no more category/level). 
Both of the scenarios has its pros and cons. 

Originally the default behavior was to drop the levels, then it adressed some issues in madshapR (grouping variables were cast to factor, then unfactored somewhere else in the function summarize()). So, to avoid that, the default behavior was to keep them (a factor is intrinsic to a variable, whether there is a dictionary or not).

Then it consequently adressed other issues in Rmonize (variable to be harmonize kept their factor status, so it is harder to work with it, such as with floor(var) function). But the user shouldn't be bothered with such problems (a factor is intrinsic to R objects and not the values, and the harmonization should work whether the column is a factor or not. Only the value matters)

So to make peace with both scenarios, a parameter **zap_factor** has been added (working similarly with stringAsFactor = TRUE/FALSE) which preserves (by defaut) the factor, make it work everywhere in both packages ; 

<image src=https://github.com/user-attachments/assets/d657f473-9ad4-4b2a-b6d1-b6a3c3bc3f7e width=60% display = block>

and allows to drop them (internally) in the function harmo process. 

<image src=https://github.com/user-attachments/assets/29101b06-0023-4902-af1b-392ce86a9335 width=50% display=block>

**Comment by twey2** [comment]
This error has been fixed in my most recent check. (Package versions Rmonize_1.1.0.9107, madshapR_1.1.0.9115, fabR_2.1.0.9103)
But, lubridate must be loaded otherwise there is an error in study5 sdc_age_m algorithm. Can be fixed by changing algorithm to use "lubridate::years".

**Comment by GuiFabre** [comment]
Yes, lubridate must be loaded to make the dpe work. This is why it appears in the Suggests, and in 'library(lubridate)' is explicitly called in the example.

thanks !

**Comment by twey2** [comment]
I think it would be better to change "years" to "lubridate::years" in the algorithm anyway? The algorithm does already use "lubridate::interval".

**Comment by GuiFabre** [comment]
yes, sure !

**Comment by twey2** [comment]
Additional comment: With the latest testing-2 versions (Rmonize_1.1.0.9200, madshapR_1.1.0.9200, fabR_2.1.0.9103), I had to load package 'car' separately. Since this is used in harmo_process(), shouldn't it be loaded automatically?

**Comment by GuiFabre** [comment]
Hello @twey2, 
following the conversation with this [issue](https://github.com/maelstrom-research/Rmonize/issues/70),
There is no "win win" situation on this one.

**Either the package car is automatically loaded** as a dependency of the package, **but the user gets the message that there is a conflict with the function dplyr::recode** and the function car::recode, each time the package is loaded.

**Or the package is in the suggestions,** with a message when the harmonization rule "recode" is used (one message, saying "car" must be downloaded to make it work, as you mentionned, **but the user must do it separately.**

Either way, we cannot avoid using dplyr function, which is widely used in the pacakage across quite all functions (dplyr::mutate to only mention one)

Using car::recode function rather dplyr::recode was easier at a time, because the structure of the car::recode function is very similar to our function "recode", and dplyr::recode is not.
At the end of the day, the best move would be to re-write the function "recode" of our package, avoiding using car::recode function. Which is feasible, but not for this version.
